### DISCUSSION TITLE
Unified issue tracker between HTTPX and HTTPCore

### DISCUSSION THREAD

**Discussion by florimondmanca** [encode/httpx#1528] [General]
HTTPX and [HTTPCore](https://github.com/encode/httpcore) are companion projects - one does high-level client features, the other low-level HTTP networking.

Currently we have two separate bug trackers, one on each repo. At the beginning this seemed natural, since there's a clear responsibility split between the two projects, so surely issues can be split based on this frontier as well.

However in practice it induces a bit of pain in terms of contributor and maintainer experience. Folks tend to open issues on HTTPX that may belong to HTTPCore, and transfering them means that those issues aren't directly visible by people that visit the HTTPX repo (because, well, where else would they go to?).

@tomchristie raised the idea of using a single issue tracker (the one on HTTPX), and I think it might make sense.

We can add an `httpcore` label for any issue that actually relates to HTTPCore rather than HTTPX. Everything will be visible by everyone in one place. Since we can close issues from PRs across repos this won't hinder the contributor experience on the HTTPCore side either.

Any further thoughts?

**Comment by lovelydinosaur** [comment]
At the moment it feels like the right thing, yup.

I'm not *even* against us thinking about if we should just be maintaining it all as a single package, but with a really clearly defined interface split at the transport API level.

* Pros: Easier release management our side, less potential for versioning issues when users report issues / trying to figure out when something was fixed/broken etc.
* Cons: Not having a strict package split makes it less obvious to users/devs that each part can be treated strictly in isolation, when trying to understand things under the hood. Easier for us to fall into bad separation of concerns. Perhaps harder to convince other projects to adopt our transport layer?

**Comment by florimondmanca** [comment]
@tomchristie Dunno about the package split, I think it really nicely enforces the interface split (eg by making it harder to sneak network features in HTTPX and vice versa). But that's something we can discuss separately if we feel it. :-)

So as for unifying issue trackers on this HTTPX repo, a clear yes it seems? Curious what the migration process would look like. I guess something like this?

* [x] Create an `httpcore` label in this repo.
* [ ] Transfer issues from HTTPCore to this repo. - Example here: https://github.com/encode/httpx/issues/1245
* [ ] Add a note in the HTTPCore README that the issue tracker is located on the HTTPX repo.
* [ ] Disable issues in the HTTPCore repo.

**Comment by florimondmanca** [comment]
I transfered https://github.com/encode/httpx/issues/1245 here and applied the `httpcore` label on it. Does this feel enough for clearly identifying what belongs to HTTPX, and what belongs to HTTPCore?
